src/Core/Environment.php

```diff
@@ -187,16 +187,22 @@ public static function getTimeLimitMax()
      */
     public static function getEnv($name)
     {
-        switch (true) {
-            case  is_array(static::$env) && array_key_exists($name, static::$env):
-                return static::$env[$name];
-            case  is_array($_ENV) && array_key_exists($name, $_ENV):
-                return $_ENV[$name];
-            case  is_array($_SERVER) && array_key_exists($name, $_SERVER):
-                return $_SERVER[$name];
-            default:
-                return getenv($name);
+        if (array_key_exists($name, static::$env)) {
+            return static::$env[$name];
         }
+        // isset() is used for $_ENV and $_SERVER instead of array_key_exists() to fix a very strange issue that
+        // occured in CI running silverstripe/recipe-kitchen-sink where PHP would timeout due apparently due to an
+        // excessively high number of array method calls. isset() is not used for static::$env above because
+        // values there may be null, and isset() will return false for null values
+        // Symfony also uses isset() for reading $_ENV and $_SERVER values
+        // https://github.com/symfony/dependency-injection/blob/6.2/EnvVarProcessor.php#L148
```

Comment:
Does setting `MY_VAR=null` and `MY_VAR=` in .env still set the value to `null`? If not, this is a breaking change.

Reply:
Just tested, they're both null